### PR TITLE
WFLY-12511 Update community documentation for Jakarta EE 8

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.jboss.org/browse/WFLY-12511
First pass of doc changes for Jakarta EE 8

### CODE REVIEW

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/EE_Concurrency_Utilities_Configuration.adoc [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [[EE_Concurrency_Utilities_Configuration]]
 = EE Concurrency Utilities
 
-EE Concurrency Utilities (JSR 236) were introduced with Java EE 7, to
```
Comment (scottmarlow) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/EE_Concurrency_Utilities_Configuration.adoc]: This should either remain as Java EE 7 or perhaps change to "EE Concurrency Utilities were introduced to ease the task of writing multithreaded Jakarta EE applications."  @emmartins what do you prefer?

Comment (bstansberry) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/EE_Concurrency_Utilities_Configuration.adoc]: +1 to just dropping "Java EE 7".

Tangent: IMHO in general this kind of "introduced with XXX" text sprinkled through docs is more maintenance trouble than it is worth. Typically XXX is "WildFly <some_number> but we see here that even Java EE X adds some cost.

File: docs/src/main/asciidoc/_developer-guide/EJB_invocations_from_a_remote_client_using_JNDI.adoc [hunk 2]
```diff
@@ -332,7 +332,7 @@ name are as follows:
 *app-name* : This is the name of the .ear (without the .ear suffix) that
 you have deployed on the server and contains your EJBs.
 
-* Java EE 6 allows you to override the application name, to a name of
+* Jakarta EE 8 allows you to override the application name, to a name of
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_developer-guide/EJB_invocations_from_a_remote_client_using_JNDI.adoc]: WDYT about just dropping the '8' here and in most of the similar places where you're doing this kind of version correction?

If EE ever evolves such that the doc chunks with these #s included are no longer spec-required etc, there would need to be a doc change regardless.  I doubt having #s in the text in some spots would help much in finding places to correct, so I don't think the #s add much value.

Of course there are places where the # is required, e.g. anything where we described what spec version we support.

**Comment by bstansberry** [comment]
@scottmarlow Please ping me when this is ready to go.

File: docs/src/main/asciidoc/_developer-guide/How_do_I_migrate_my_application_from_AS7_to_WildFly.adoc [hunk 3]
```diff
@@ -16,13 +16,13 @@ The purpose of this guide is to document changes that are needed to successfully
 
 The list of {ProductName} new functionality is extensive, being the most relevant, with respect to server and application migrations:
 
-* Java EE7 - {ProductName} is a certified implementation of Java EE7, meeting both the Web and the Full profiles, and already includes support for the latest iterations of CDI (1.2) and Web Sockets (1.1).
+* Jakarta EE 8 - {ProductName} is a certified implementation of Jakarta EE 8, meeting both the Web and the Full profiles, and already includes support for the latest iterations of CDI (1.2) and Web Sockets (1.1).
```
Comment (jamezp) [inline docs/src/main/asciidoc/_developer-guide/How_do_I_migrate_my_application_from_AS7_to_WildFly.adoc]: Full is not called "Jakarta EE 8 Full Platform" and the web is still called "Jakarta EE 8 Web Profile". We may need to change the full reference(s) to Platform.

File: ee/src/main/java/org/jboss/as/ee/concurrent/ManagedScheduledExecutorServiceImpl.java [hunk 4]
```diff
@@ -38,7 +38,7 @@
 import static org.jboss.as.ee.concurrent.ControlPointUtils.doWrap;
 
 /**
- * WildFly's extension of Java EE 7 RI {@link org.glassfish.enterprise.concurrent.ManagedScheduledExecutorServiceImpl}.
+ * WildFly's extension of Jakarta EE 7 RI {@link org.glassfish.enterprise.concurrent.ManagedScheduledExecutorServiceImpl}.
```
Comment (jamezp) [inline ee/src/main/java/org/jboss/as/ee/concurrent/ManagedScheduledExecutorServiceImpl.java]: The version here should likely be removed.

**Review by jamezp — COMMENTED** [review]
I think for the most part this looks okay. I do think we should remove the version number, e.g. 8, where we can or at least change it to use the property `{javaee_version}` which should also likely be renamed.

I also think we should remove the various API versions where we can. I think some of them are wrong anyway. There is a reference to JAX-RS 2.0 and we're at 2.1.

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/EE_Concurrency_Utilities_Configuration.adoc (hunk 1 above)
Comment (scottmarlow) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/EE_Concurrency_Utilities_Configuration.adoc]: Updating to:

> EE Concurrency Utilities (JSR 236) were introduced to ease the task of writing multithreaded applications. Instances of these utilities are managed by WildFly, and the related configuration.

File: docs/src/main/asciidoc/_developer-guide/EJB_invocations_from_a_remote_client_using_JNDI.adoc (hunk 2 above)
Comment (scottmarlow) [reply docs/src/main/asciidoc/_developer-guide/EJB_invocations_from_a_remote_client_using_JNDI.adoc]: makes sense, thanks.

Comment (scottmarlow) [reply docs/src/main/asciidoc/_developer-guide/EJB_invocations_from_a_remote_client_using_JNDI.adoc]: just updated locally, will push changes soon, commits should be squashed but will leave separate for now.

File: docs/src/main/asciidoc/Getting_Started_Guide.adoc [hunk 5]
```diff
@@ -54,7 +54,7 @@ Profile |WildFly {wildflyVersion} Full Profile |WildFly {wildflyVersion} Web Pro
 
 |JSR-352: Batch Applications for the Java Platform 1.0 |X |-- |X |--
 
-|JSR-236: Concurrency Utilities for Java EE 1.0 |X |X |X |X
+|JSR-236: Concurrency Utilities for Jakarta EE |X |X |X |X
```
[the remark this answers is not in the document]
Comment (scottmarlow) [reply docs/src/main/asciidoc/Getting_Started_Guide.adoc]: Thanks James, I just removed the others also.

File: docs/src/main/asciidoc/Getting_Started_Guide.adoc [hunk 6]
```diff
@@ -289,26 +289,26 @@ process.
 ===== Standalone Server Configurations
 
 * standalone.xml (_default_)
-** Java Enterprise Edition {javaee_version} web profile certified configuration with
+** Jakarta Enterprise Edition {javaee_version} web profile certified configuration with
```
[the remark this answers is not in the document]
Comment (scottmarlow) [reply docs/src/main/asciidoc/Getting_Started_Guide.adoc]: I will change:
  Jakarta Enterprise Edition
To:
Jakarta EE

Good catch, there are two of these.

[the remark this answers is not in the document]
Comment (scottmarlow) [reply docs/src/main/asciidoc/Getting_Started_Guide.adoc]: Also removing  {javaee_version}

File: docs/src/main/asciidoc/_developer-guide/How_do_I_migrate_my_application_from_AS7_to_WildFly.adoc (hunk 3 above)
Comment (scottmarlow) [reply docs/src/main/asciidoc/_developer-guide/How_do_I_migrate_my_application_from_AS7_to_WildFly.adoc]: Good catch, updating to "Full Profile".  Platform could also work

File: ee/src/main/java/org/jboss/as/ee/concurrent/ManagedScheduledExecutorServiceImpl.java (hunk 4 above)
Comment (scottmarlow) [reply ee/src/main/java/org/jboss/as/ee/concurrent/ManagedScheduledExecutorServiceImpl.java]: I think that we can also remove Jakarta, will change to:
>  WildFly's extension of {@link org.glassfish.enterprise.concurrent.ManagedScheduledExecutorServiceImpl}.

**Comment by scottmarlow** [comment]
I'll change JAX-RS 2.0, to just JAX-RS and push changes.  

I think that we need at least one more pass to check if there are new Jakarta names for things like JAX-RS, which according to https://www.eclipse.org/lists/jaxrs-dev/msg00680.html should be "Jakarta RESTful Web Services" or something like that.  

We should use https://jakarta.ee/legal/acronym_guidelines/ as a guide.

**Comment by scottmarlow** [comment]
Do we really need to change from "EJB" to "Jakarta Enterprise Beans"?  We will discuss this elsewhere and update this pull request after.

**Comment by scottmarlow** [comment]
We will start a separate branch for renaming JAX-RS + EJB + many others, will update this pr to not include the JAX-RS (cleanup) and EJB changes.

**Comment by scottmarlow** [comment]
@bstansberry @jamezp the changes in this pr are now complete.

**Comment by scottmarlow** [comment]
dependency tree error is:

> Canceled (Error while applying patch; cannot find commit 00cf8e2db397753b4b5beca60618f5d4b6a4d345 in the https://github.com/wildfly/wildfly.git repository, possible reason: refs/pull/12604/merge branch was updated and the commit selected for the buil...
> Canceled with comment: Build revision not found 

need to figure out the cause of that

**Comment by scottmarlow** [comment]
This is a common failure that others have seen

**Review by bstansberry — APPROVED** [review]

**Pull request merged**